### PR TITLE
SD-1471 Geo API - Add States endpoints to Platform API

### PR DESCRIPTION
**Pull request by aleksandarpetrushev**
https://spark-solutions.atlassian.net/browse/SD-1471

### CODE REVIEW

**Review by damianlegawiec — COMMENTED** [review]
@aleksandarpetrushev  you need to run rswag to generate documentation, also let's add an example of how to filter state list by country and include country

File: api/spec/integration/api/v2/platform/states_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    end
+  end
+
+  path '/api/v2/platform/states/{id}' do
+    let(:id) { create(:state).id }
+
+    get 'Returns a State' do
+      tags 'States'
+      security [ bearer_auth: [] ]
+      operationId 'show-state'
```
Comment (damianlegawiec) [inline api/spec/integration/api/v2/platform/states_spec.rb:25]: here we have show-state, and list if states-list - it's inconsistent

Comment (aleksandarpetrushev) [reply api/spec/integration/api/v2/platform/states_spec.rb:25]: I do not follow, we have that for other resources: addresses-list & show-address etc.
In the shared examples: operationId "#{resource_name.parameterize.pluralize.to_sym}-list" & "show-#{resource_name.parameterize.to_sym}"

File: core/app/models/spree/state.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ class State < Spree::Base
     validates :country, :name, presence: true
     validates :name, :abbr, uniqueness: { case_sensitive: false, scope: :country_id }, allow_blank: true
 
-    self.whitelisted_ransackable_attributes = %w(abbr)
+    self.whitelisted_ransackable_attributes = %w(abbr country_id)
```
Comment (damianlegawiec) [inline core/app/models/spree/state.rb:17]:
```suggestion
    self.whitelisted_ransackable_attributes = %w(abbr country_id name)
```

**Pull request merged**